app/src/test/kotlin/compiler/e2e/ParserErrorsTest.kt

```diff
@@ -1,5 +1,441 @@
+                            
+                        """,
+                """
+                            czynność f() {
+                                dopóki () { }
+                            }
+                            
+                        """,
+                """
+                            czynność f() {
+                                dopóki (zwróć fałsz) { }
+                            }
+                            
+                        """,
+            )
+        )
+    }
+
+    @Test
+    fun `test return`() {
+        assertParseError(
+            listOf(
+                """
+                            czynność f() -> Liczba {
+                                return 5
+                            }
+                            
+                        """,
+                """
+                            czynność f() -> Liczba {
+                                zwróć zm test: Liczba = 17
+                            }
+                            
+                        """,
+                """
+                            czynność f() -> Liczba {
+                                zakończ 17
+                            }
+                            
+                        """,
+                """
+                            czynność f() {
+                                zakończ program
+                            }
+                            
+                        """,
+                """
+                            czynność f() {
+                                zwróć
+                            }
+                            
+                        """,
+                """
+                            czynność f() {
+                                zwróć Nic
+                            }
+                            
+                        """,
+            )
+        )
+    }
+
+    @Test
+    fun `test colons`() {
+        assertParseError(
+            listOf(
+                """
+                            zm test: Liczba = 17;;
+                            
+                        """,
+                """
+                            zm test; Liczba = 17
+                            
+                        """,
+                """
+                            czynność f(a: Liczba; b: Liczba) { }
+                            
+                        """,
+                """
+                            czynność f(a: Liczba,, b: Liczba) { }
+                            
+                        """,
+                """
+                            czynność f(a; Liczba) { }
+                            
+                        """,
+            )
+        )
+    }
+
+    @Test
+    fun `test expressions`() {
+        assertParseError(
+            listOf(
+                """
+                            czynność f(x: Liczba) -> Liczba {
+                                zwróć x
+                                + x
+                            }
```

Comment:
My bad, but we actually have the unary plus operator, so this one is correct after all